radicle-cli/src/commands/rm.rs

```diff
@@ -0,0 +1,187 @@
+pub struct Options {
+    object: Object,
+    confirm: bool,
+    passphrase: bool,
+    stdin: bool,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut object: Option<Object> = None;
+        let mut confirm = true;
+        let mut passphrase = true;
+        let mut stdin = false;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("no-confirm") => {
+                    confirm = false;
+                }
+                Long("no-passphrase") => {
+                    passphrase = false;
+                }
+                Long("stdin") => {
+                    stdin = true;
+                }
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Value(val) if object.is_none() => {
+                    let val = val.to_string_lossy();
+                    let val = Object::from(val.as_ref());
+                    object = Some(val);
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((
+            Options {
+                object: object.ok_or_else(|| {
+                    anyhow!("Urn or peer id to remove must be provided; see `rad rm --help`")
+                })?,
+                confirm,
+                passphrase,
+                stdin,
+            },
+            vec![],
+        ))
+    }
+}
+
+pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    term::warning("Experimental tool; use at your own risk!");
+
+    let profile = ctx.profile()?;
+    let storage = &profile.storage;
+    let signer = term::signer(&profile)?;
+
+    match &options.object {
+        Object::Project(id) => {
+            if let Ok(Some(_)) = storage.get(signer.public_key(), id.to_owned()) {
+                let monorepo = profile.paths().storage();
+                let namespace = monorepo
+                    .join(&id.to_human())
+                    .join("refs")
+                    .join("namespaces");
+
+                if !options.confirm
+                    || term::confirm(format!(
+                        "Are you sure you would like to delete {}?",
+                        term::format::dim(namespace.display())
+                    ))
+                {
+                    super::rad_untrack::untrack(id.to_owned(), &profile)?;
+                    fs::remove_dir_all(namespace)?;
+                    term::success!("Successfully removed project {}", &id);
+                }
+            } else {
+                anyhow::bail!("project {} does not exist", &id)
+            }
+        }
+        Object::User(pk) => {
+            //let profiles = profile::list()?;
+            //if storage.peer_id() != peer_id {
+            if signer.public_key() != pk {
```

Comment:
since we only have 1 profile per rad directory - the one in signer - this if would never evaluate to `true`.

the alternative would seem to be to skip the check, and totally remove the whole directory. is this what we want? (since profile = pk = dir)

Reply:
Yeah basically this command should only delete projects, not users.